Content.Shared/Machines/Components/MultipartMachineComponent.cs

```diff
@@ -0,0 +1,103 @@
+using Content.Shared.Machines.EntitySystems;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Machines.Components;
+
+/// <summary>
+/// Marks an entity as being the owner of a multipart machine.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+[Access(typeof(SharedMultipartMachineSystem))]
+public sealed partial class MultipartMachineComponent : Component
+{
+    /// <summary>
+    /// Dictionary of Enum values to specific parts of this machine.
+    /// Each key can be specified as 'enum.<EnumName>.<EnumValue>` in Yaml.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public Dictionary<Enum, MachinePart> Parts = [];
+
+    /// <summary>
+    /// Whether this multipart machine is assembled or not.
+    /// Optional parts are not taken into account.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool IsAssembled = false;
+
+    /// <summary>
+    /// Flag for whether the client side system is allowed to show
+    /// ghosts of missing machine parts.
+    /// Controlled/Used by the client side.
+    /// </summary>
+    [DataField]
+    public List<EntityUid> Ghosts = [];
```

Comment:
If only the client uses these then delete datafield as client won't serialize them.

If server uses then needs autonetworkedfield.